core/app/models/spree/wallet_payment_source.rb

```diff
@@ -1,19 +1,31 @@
 # frozen_string_literal: true
 
-class Spree::WalletPaymentSource < ActiveRecord::Base
-  belongs_to :user, class_name: Spree::UserClassHandle.new, foreign_key: 'user_id', inverse_of: :wallet_payment_sources
-  belongs_to :payment_source, polymorphic: true, inverse_of: :wallet_payment_sources
+module Spree
+  class WalletPaymentSource < Spree::Base
+    belongs_to :user, class_name: Spree::UserClassHandle.new, foreign_key: 'user_id', inverse_of: :wallet_payment_sources
+    belongs_to :payment_source, polymorphic: true, inverse_of: :wallet_payment_sources
 
-  validates_presence_of :user
-  validates_presence_of :payment_source
+    validates_presence_of :user
+    validates_presence_of :payment_source
+    validates :user_id, uniqueness: { scope: [:payment_source_type, :payment_source_id] }
+    validate :check_for_payment_source_class
```

Comment:
I'm guessing the reasoning for this is to mimic what happens in the `Spree::Wallet#add` method? I'm a fan prepending validation methods with `validate` so you know the purpose of that method is to validate and if called will add to the object errors. `validate_payment_source_class`

Reply:
@skukx agree but these validations were already there, they have just been indented due to the module change. I think that if we want to change them we need to open another PR and deprecate existing methods.